http-ballerina/http_request.bal

```diff
@@ -389,6 +389,8 @@ public class Request {
                             string value = entries[entryIndex].substring(index + 1, size);
                             value = value.trim();
                             if (value != "") {
+                                name = check decode(name, CHARSET_UTF_8);
+                                value = check decode(value, CHARSET_UTF_8);
```

Comment:
* Do we have to do url-decode for each entry? Can't we just decode the whole payload once and then split it with `&` ?